sys/linux/vnet.txt

```diff
@@ -738,24 +736,117 @@ ipv6_payload [
 # http://www.iana.org/assignments/tcp-parameters/tcp-parameters.xhtml
 
 include <net/tcp.h>
+include <net/mptcp.h>
+include <net/mptcp_v4.h>
+include <net/mptcp_v6.h>
 include <uapi/linux/tcp.h>
 
 tcp_option [
-	generic		tcp_generic_option
-	nop		tcp_nop_option
-	eol		tcp_eol_option
-	mss		tcp_mss_option
-	window		tcp_window_option
-	sack_perm	tcp_sack_perm_option
-	sack		tcp_sack_option
-	timestamp	tcp_timestamp_option
-	md5sig		tcp_md5sig_option
-	fastopen	tcp_fastopen_option
-	exp_fastopen	tcp_exp_fastopen_option
-	exp_smc		tcp_exp_smc_option
+    generic		tcp_generic_option
+    nop		tcp_nop_option
+    eol		tcp_eol_option
+    mss		tcp_mss_option
+    window		tcp_window_option
+    sack_perm	tcp_sack_perm_option
+    sack		tcp_sack_option
+    timestamp	tcp_timestamp_option
+    md5sig		tcp_md5sig_option
+    fastopen	tcp_fastopen_option
+    exp_fastopen	tcp_exp_fastopen_option
+    exp_smc	tcp_exp_smc_option
+    mptcp       tcp_mptcp_option
+] [varlen]
+
+tcp_mptcp_option [
+    generic     mptcp_generic_option
+    syn         mptcp_syn_option
+    synack      mptcp_synack_option
+    ack         mptcp_ack_option
+    capable     mptcp_capable_option
+    add_addr    mptcp_add_addr_option
+    mp_join     mptcp_mp_join_option
+    mp_flcose   mptcp_mp_fclose_option
+    remove_addr mptcp_remove_addr_option
```

Comment:
This seems to miss the Fallback (MP_FAIL) Option.

Reply:
i removed that option because it is not listed in mptcp.h as option although it has its own struct in mptcp.h. but will add this

Reply:
If it's not in the code being fuzzed, I guess there's no reason to add it to the descriptions, so consider this comment resolved. Could you share a link to the code btw?

Reply:
which code? containing the mptcp.h?

Reply:
Yes.

Reply:
[](https://github.com/multipath-tcp/mptcp_net-next/blob/master/include/net/mptcp.h)